libcaf_core/src/logger.cpp

```diff
@@ -223,7 +247,9 @@ void logger::log(int level, const char* component,
                  const std::string& class_name, const char* function_name,
                  const char* c_full_file_name, int line_num,
                  const std::string& msg) {
-  CAF_ASSERT(level >= 0 && level <= 4);
+  CAF_ASSERT(level <= 4);
```

Comment:
That was a left-over. I temporarily switch the level to `size_t`. Will undo that change.